src/comparator.ml

```diff
@@ -0,0 +1,8 @@
+
+type 'a t = 'a -> 'a -> TotalOrder.t
+
+let make (compare : 'a -> 'a -> TotalOrder.t) (this : 'a) (that : 'a): TotalOrder.t =
```

Comment:
I took it from [here](https://github.com/facebookincubator/immutable-re/blob/master/immutable/src/core/Comparator.re#L12). I guess the intention is provide a way to wrap a simple comparison function into a `Comparator.t`, but I can remove it if you think it is too trivial.